hooks/pre/10-reset_feature.rb

```diff
@@ -1,32 +1,45 @@
 # See bottom of the script for the command that kicks off the script
 
 def reset
-  if foreman_installed?
-
-    stop_services
-    reset_database
-    reset_candlepin
-    reset_pulp
-    reset_elasticsearch
+  stop_services
+  reset_database
+  reset_candlepin
+  reset_pulp
+  reset_elasticsearch
+end
 
+def rake(task)
+  if Kafo::Helpers.command_exists?('foreman-rake')
+    command = "foreman-rake #{task}"
   else
-    Kafo::KafoConfigure.logger.warn 'Katello not installed yet, can not drop database!'
+    user = param('katello_devel', 'user').value
+    rvm = param('katello_devel', 'rvm_ruby').value
+    command = "sudo su #{user} -c '/bin/bash --login -c \"rvm use #{rvm} && bundle exec rake #{task}\"'"
   end
+
+  command
 end
 
-def foreman_installed?
-  `which foreman-rake > /dev/null 2>&1`
-  $?.success?
+def service
+  Kafo::Helpers.command_exists?('service-wait') ? 'service-wait' : 'service'
 end
 
 def stop_services
   Kafo::KafoConfigure.logger.info 'Ensuring services httpd and foreman-tasks are stopped.'
-  Kafo::Helpers.execute('service httpd stop && service foreman-tasks stop')
+  Kafo::Helpers.execute("#{service} httpd stop")
+  Kafo::Helpers.execute("#{service} foreman-tasks stop") if Kafo::Helpers.service_exists?('foreman-tasks')
```

Comment:
Can we just use the new katello-service tool? It'll stop whatever's available, `katello-service --exclude mongod,postgresql` I think would do what you want and all the service commands could be removed from the other sections.